sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JsonUtils.java

```diff
@@ -86,13 +85,39 @@ public static <T> JsonWriter writeArray(JsonWriter jsonWriter, String fieldName,
         return jsonWriter.writeEndArray().flush();
     }
 
+    /**
+     * Serializes a map.
+     *
+     * @param jsonWriter The {@link JsonWriter} where JSON will be written.
+     * @param fieldName Field name for the map.
+     * @param map The map.
+     * @param entryWriterFunc Function that writes the map entry value.
+     * @param <T> Type of map value.
+     * @return The updated {@link JsonWriter} object.
+     */
+    public static <T> JsonWriter writeMap(JsonWriter jsonWriter, String fieldName, Map<String, T> map,
+        BiConsumer<JsonWriter, T> entryWriterFunc) {
```

Comment:
Should APIs that work on jdk and template types be on `JsonWriter` ?
I.e. should this be `JsonWriter.writeMap(String fieldName, Map<String, T> map,
        BiConsumer<JsonWriter, T> entryWriterFunc)` ?
        
After reading a bit more.
Perhaps current `JsonWriter` is `JsonCodec`/`JsonTokenWriter`/`JsonTokenizer` ? and `JsonUtils` could become `JsonWriter`/`JsonReader`? I.e. can we replace statics with layer of higher level types?

Reply:
I think it'll be a good GA check to determine which of the utility methods should be pushed back into `JsonWriter`/`JsonReader`. I'll add that to the list of follow-ups